src/mqtt.h

```diff
@@ -7,53 +7,75 @@
 
 #pragma once
 
+#include "Parameter.h"
 #include "userConfig.h"
 #include <Arduino.h>
 #include <PubSubClient.h>
 #include <os.h>
 
-unsigned long previousMillisMQTT;
-const unsigned long intervalMQTT = 5000;
+inline unsigned long previousMillisMQTT;
+constexpr unsigned long intervalMQTT = 5000;
 
-WiFiClient net;
-PubSubClient mqtt(net);
+inline WiFiClient net;
+inline PubSubClient mqtt(net);
 
-const char* mqtt_server_ip = MQTT_SERVER_IP;
-const int mqtt_server_port = MQTT_SERVER_PORT;
-const char* mqtt_username = MQTT_USERNAME;
-const char* mqtt_password = MQTT_PASSWORD;
-const char* mqtt_topic_prefix = MQTT_TOPIC_PREFIX;
+inline bool mqtt_enabled = false;
+inline String mqtt_server_ip = "";
+inline int mqtt_server_port = 1883;
+inline String mqtt_username = "";
+inline String mqtt_password = "";
+inline String mqtt_topic_prefix = "";
+inline bool mqtt_hassio_enabled = false;
+inline String mqtt_hassio_discovery_prefix = "";
 
-char topic_will[256];
-char topic_set[256];
+inline char topic_will[256];
+inline char topic_set[256];
 
-unsigned long lastMQTTConnectionAttempt = millis();
-unsigned int MQTTReCnctCount = 0;
+inline unsigned long lastMQTTConnectionAttempt = millis();
+inline unsigned int MQTTReCnctCount = 0;
 
-extern std::map<const char*, std::function<editable_t*()>, cmp_str> mqttVars;
+extern std::map<const char*, const char*, cmp_str> mqttVars;
 extern std::map<const char*, std::function<double()>, cmp_str> mqttSensors;
 
 struct DiscoveryObject {
         String discovery_topic;
         String payload_json;
 };
 
+inline void setupMqtt() {
+    ParameterRegistry& registry = ParameterRegistry::getInstance();
+
+    if (!registry.isReady()) {
+        LOGF(ERROR, "ParameterRegistry not ready, cannot initialize MQTT");
+        return;
+    }
+
+    mqtt_enabled = registry.getParameterById("MQTT_ENABLED")->getBoolValue();
```

Comment:
should have noted above, but you could template the function directly to make this

```cpp
    mqtt_enabled = registry.get<bool>("MQTT_ENABLED");
```